spec/integration/ohai/ohai_spec.rb

```diff
@@ -0,0 +1,56 @@
+require "spec_helper"
+require "chef/mixin/shell_out"
+
+describe "ohai" do
+  include Chef::Mixin::ShellOut
+
+  let(:ohai) { "bundle exec ohai" }
+
+  describe "testing ohai performance" do
+    # The purpose of this test is to generally find misconfigured DNS on
+    # CI testers.  If this fails, it is probably because the forward+reverse
+    # DNS lookup that node[:hostname] needs is timing out and failing.
+    #
+    # If it is failing spuriously, it may mean DNS is failing spuriously, the
+    # best solution will be to make sure that `hostname -f`-like behavior hits
+    # /etc/hosts and not DNS.
+    #
+    # If it still fails supriously, it is possible that the server has high
+    # CPU load (e.g. due to background processes) which are contending with the
+    # running tests (disable the screensaver on servers, stop playing Fortnite
+    # while you're running tests, etc).
+    #
+    # If this just fails due to I/O being very slow and ruby being very slow to
+    # startup then that still indicates that the tester configuration needs
+    # fixing.  The fact that this will fail on a windows box on a virt that doesn't
+    # use an SSD is because we have a higher bar for the tests to run successfully
+    # and that configuration is broken, so this test is red for a reason.
+    #
+    # This will probably fail on raspberry pi's or something like that as well.  That
+    # is not a bug.  We will never accept a raspberry pi as a CI tester for our
+    # software.  Feel free to manually delete and thereby skip this file in your
+    # own testing harness, but that is not our concern, we are testing behavior
+    # that is critical to our infrastructure and must run in our tests.
+    #
+    it "the hostname plugin must return in under 4 seconds" do
```

Comment:
Would it make sense to say "see code comment in case of failure". You have a pretty nice comment up there but if this fails in 2 years we're going to miss it

Reply:
tbh I'd be fine with printing the whole comment if this is expected to fail very rarely if ever.

Reply:
this test probably doesn't work, if i do anything it'll probably be the other test.

Reply:
(windows is just way too slow normally)

Reply:
(although maybe autoloading and related approaches would help a lot)